### DISCUSSION TITLE
Is it possible to read my encrypted bookmarks without Floccus if I have the encryption password ?

### DISCUSSION THREAD

**Discussion by pinpontitit** [floccusaddon/floccus#1149] [Q&A] [answered]
Sorry if the answer is obvious but I am not a professional of encryption.

Currently I use google drive to store my encrypted file containing my bookmarks.

Imagine that one day, Floccus disappears and all the other similar apps also, and nobody, including myself, is able to recreate a fork of Floccus.

But i still remember my encryption password.

Would it be possible to decrypt my file ?
And how could I do that ?

Because, since the encryption protocol used (AES256 ?) is a widespread "protocol", I was thinking that some software existed that could work like this:

I have an Aes256 encrypted file, I have a password, I put those two data into the software, and it gives me the unencrypted version of the file.

Does it work like that ?
Does such software or solution exists ?

thanks for learning me someting

**Comment by mnalis** [comment, marked as answer]
To decrypt something, one needs to know:

- exact encryption protocol used (e.g. not just `AES` but its exact [subvariant](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation) - is it `CBC`, or `GCM`, or `SIV`, or...)
- all non-secret variables used by protocol (e.g. that `256` is block size, some versions use initialization vector, etc.)
- secret itself (i.e. passphrase)
- data container information (e.g. is  encrypted data stored raw in file, or does file itself has some headers, paddings, etc. - or is it even some sort of database like `sqlite` etc).

If you know all that (and you do if you have floccus running on any browser, as it is open source), then you're in luck and you can decrypt your bookmarks.  Maybe you could even use available tools in some cases (e.g. `openssl list -cipher-commands`) with minimal or no shell scripting, but most likely you'd have to write a small program to recover your information.


However, if **not** all of that information is known to you, and all versions of floccus have been lost, you can try to detect (e.g. on GNU/Linux systems, command `file(1)` might give you information on some popular containers) or guess (e.g. try most popular encryption versions with most popular parameters etc. and hope for the best while checking results manually one by one). 

While not impossible, that however would much more likely resemble amateur cracking attempts (or even more serious cryptoanalyses) than regular data recovery, and is extremely likely to be **unsuccessful** unless much knowledge and resources are thrown at the problem (which is unlikely to happen in real life).

So I suggest [looking into source right now](https://github.com/floccusaddon/floccus/blob/develop/src/lib/Crypto.ts) - it is less than 100 lines. Just in case, if civilization ends and wipes all copies of floccus :smiley: , you'd maybe at least remember the majority of variables. Or just [download](https://github.com/floccusaddon/floccus/archive/refs/heads/develop.zip) the source to several of your computers and put it in your backups.
